src/Workspaces/CSharp/Portable/CodeGeneration/CSharpCodeGenerationService.cs

```diff
@@ -482,6 +482,21 @@ protected override TDeclarationNode AddMembers<TDeclarationNode>(TDeclarationNod
             {
                 return (accessorDeclaration.Body == null) ? destinationMember : Cast<TDeclarationNode>(accessorDeclaration.AddBodyStatements(StatementGenerator.GenerateStatements(statements).ToArray()));
             }
+            else if (destinationMember is CompilationUnitSyntax compilationUnit && options is null)
```

Comment:
why the check for `options is null`?

Reply:
This method had an existing caller that passed in `CompilationUnitSyntax` with options, and relied on the `AddStatementsWorker` fallback to handle those options. Right now I forget what caller that was.

Reply:
➡️ Will submit a follow-up to clarify the condition.